### ISSUE TITLE
Error to connect on clickhouse cloud

### ISSUE THREAD

**Issue by prettoandre** [ClickHouse/clickhouse-kafka-connect#285]
### Describe the bug

The connector is only working for connecting to a local clickhouse, if I try to connect to the clickhouse cloud the connection is not established.

### Expected behaviour
Connect to clickhouse cloud 

### Error log

![image](https://github.com/ClickHouse/clickhouse-kafka-connect/assets/1655350/60e4822b-bfa7-43c3-a41a-ead2f950b45d)

![image](https://github.com/ClickHouse/clickhouse-kafka-connect/assets/1655350/b8fcbe33-a500-43f5-8626-879540367f7e)

#### Environment
* Kafka-Connect version: clickhouse-kafka-connect:v1.0.10
* Kafka Connect configuration:

{
    "name": "ClickHouseSinkConnector_stocks_json",
    "config": {
      "connector.class": "com.clickhouse.kafka.connect.ClickHouseSinkConnector",
      "key.converter": "org.apache.kafka.connect.storage.StringConverter",
      "value.converter": "org.apache.kafka.connect.json.JsonConverter",
      "value.converter.schemas.enable": "false",
      "schemas.enable": "false",
      "tasks.max": "1",
      "errors.retry.timeout": "60",
      "errors.tolerance": "all",
      "errors.log.enable": "true",
      "exactlyOnce": "true",
      "hostname": "qd9sxpu3et.eu-west-1.aws.clickhouse.cloud",
      "username": "default",
      "password": "", 
      "ssl": "true",
      "port": "8443",
      "database": "kafka",
      "topics": "stocks",
      "auto.evolve": "false",
      "auto.create": "false"
    }
  }

* Kafka version: confluent cloud
* Kafka environment: confluent cloud
* OS: macos

#### ClickHouse server
* ClickHouse Server version: clickhouse cloud version 23.9.2
* ClickHouse Server non-default settings, if any:
* `CREATE TABLE` statements for tables involved:
* Sample data for all these tables, use [clickhouse-obfuscator](https://github.com/ClickHouse/ClickHouse/blob/master/programs/obfuscator/Obfuscator.cpp#L42-L80) if necessary

**Comment by prettoandre** [comment]
@Paultagoras  I'm setting the password on the connector to try to connect to Clickhouse Cloud but it's not working. actually Really in the local clickhouse it connects without a password.

**Comment by mshustov** [comment]
@prettoandre does your Cloud instance have any IP Filtering enabled?

**Comment by prettoandre** [comment]
> @prettoandre does your Cloud instance have IP Filtering enabled? In what env do you run the Connector?

Yes ! my instance has IP filtering, but my IP is whitelisted. When I'm using the clickhouse client command line it connects normally

**Comment by prettoandre** [comment]
> @prettoandre Hmm - how are you running the connector? Docker image?

@Paultagoras problem solved, there seems to be some incompatibility with Confluent's Kafka Connect version, we used version 7.1 and it worked. If I have time I will test more versions. 

Docker image used: https://hub.docker.com/layers/confluentinc/cp-kafka-connect/7.1.10/images/sha256-50eb9d4f56ab3f45f4e0fde47ba1e410f1f86ff7262f138ec0fb75301ad7273b?context=explore

Thank you for your help.

**Comment by batmanjo** [comment]
i get a simlar problem when using bitnami/kafka:3.7.1 , kafka connect logs say Read time out.  seems it  incompatibility with clickhouse sink connector. 

solution : finally i choose the apache/kafka:3.7.1 docker image, it work success
